src/Sylius/Behat/Context/Api/Admin/ManagingShippingCategoriesContext.php

```diff
@@ -0,0 +1,220 @@
+    }
+
+    /**
+     * @When I save my changes
+     */
+    public function iSaveMyChanges(): void
+    {
+        $this->client->update();
+    }
+
+    /**
+     * @Then I should be notified that shipping category with this code already exists
+     */
+    public function iShouldBeNotifiedThatShippingCategoryWithThisCodeAlreadyExists(): void
+    {
+        Assert::same($this->client->getError(), 'code: The shipping category with given code already exists.');
+    }
+
+    /**
+     * @Then I should be notified that :element is required
+     */
+    public function iShouldBeNotifiedThatElementIsRequired(string $element): void
+    {
+        Assert::same(
+            $this->client->getError(), sprintf('%s: Please enter shipping category %s.', $element, $element)
+        );
+    }
+
+    /**
+     * @Then I should see a single shipping category in the list
+     * @Then I should see :count shipping categories in the list
+     */
+    public function iShouldSeeShippingCategoriesInTheList(int $count = 1): void
+    {
+        $this->client->index('shipping_categories');
+        Assert::same($this->client->countCollectionItems(), $count);
+    }
+
+    /**
+     * @Then /^the (shipping category "([^"]+)") should be in the registry$/
+     * @Then /^the (shipping category "([^"]+)") should appear in the registry$/
+     */
+    public function theShippingCategoryShouldAppearInTheRegistry(ShippingCategoryInterface $shippingCategory): void
+    {
+        $shippingCategoryName = $shippingCategory->getName();
+        Assert::true(
+            $this->isItemOnIndex('name', $shippingCategoryName),
+            sprintf('Shipping category with name %s does not exists', $shippingCategoryName)
+        );
+    }
+
+    /**
+     * @Then shipping category with name :name should not be added
+     */
+    public function shippingCategoryWithNameShouldNotBeAdded(string $name): void
+    {
+        Assert::false(
+            $this->isItemOnIndex('name', $name),
+            sprintf('Shipping category with name %s exists', $name)
+        );
+    }
+
+    /**
+     * @Then /^(this shipping category) should no longer exist in the registry$/
+     */
+    public function thisShippingCategoryShouldNoLongerExistInTheRegistry(
+        ShippingCategoryInterface $shippingCategory
+    ): void {
+        $shippingCategoryName = $shippingCategory->getName();
+        Assert::false(
+            $this->isItemOnIndex('name', $shippingCategoryName),
+            sprintf('Shipping category with name %s exist', $shippingCategoryName)
+        );
+    }
+
+    /**
+     * @Then I should not be able to edit its code
+     */
+    public function iShouldNotBeAbleToEditItsCode(): void
+    {
+        $this->client->addRequestData('code', 'NEW_CODE');
+        $this->client->update();
+
+        Assert::false($this->client->responseHasValue('code', 'NEW_CODE'), 'The code field with value NEW_CODE exists');
```

Comment:
Maybe something like `The shipping category code should not be changed to "NEW_CODE", but it is`?